pkg/apis/sessions/session_state.go

```diff
@@ -46,6 +46,15 @@ func (s *SessionState) Age() time.Duration {
 	return 0
 }
 
+func (s *SessionState) AdjustExpirationByRefreshPercent(refreshPercent uint8) time.Time {
+	// Recalculate expiration based on refresh percentage
+	originalExpiresOn := s.ExpiresOn
+	newExpiresOn := s.computeSessionExpiryByPercent(s.CreatedAt, originalExpiresOn, refreshPercent)
+	s.ExpiresOn = &newExpiresOn
+
+	return *originalExpiresOn
+}
+
```

Comment:
I wouldn't expect us to be updating the underlying session's expiry, I would have expected us to use the percentage when we are checking whether the session needs refreshing or not

Reply:
There isn't really a functional difference. I considered a 2nd field of AdjustedExpiresOn due to have to be transparent in the logs when we actually refresh and when things actually expire. From the practical standpoint of how expiry is used, there isn't really a difference (tokens carry real expiry anyway).